ec2/ec2.go

```diff
@@ -787,15 +777,35 @@ type IPPerm struct {
 	Protocol     string              `xml:"ipProtocol"`
 	FromPort     int                 `xml:"fromPort"`
 	ToPort       int                 `xml:"toPort"`
-	SourceIPs    []string            `xml:"ipRanges>item>cidrIp"`
```

Comment:
Sorry, so a small correction. I tried what I suggested - running `go test -gocheck.v -gocheck.f TestIPPerms -amazon` on launchpad.net/goamz/ec2 and failed, but not due to the reason source IPs were not parsed correctly. It failed because:

1. The error code for calling AuthorizeSecurityGroupIngress with an invalid CIDR (z127.0.0.1/24) is InvalidParameterValue, not InvalidPermission.Malformed (even though the API documentation states the latter should be expected for invalid CIDR - http://docs.aws.amazon.com/AWSEC2/latest/APIReference/errors-overview.html)
2. The source IPs *get deserialized* correctly - []string{"127.0.0.0/24", "200.1.1.34/32"} (well, almost - 127.0.0.1/24 from the test is converted to the equivalent 127.0.0.0/24 by Amazon), but the length of response.Groups[0].IPPerms is 3, not 1 as expected by the test.

As some of the changes in this PR modified the failing live tests to pass. As for the SourceIPs I experimented a bit - http://play.golang.org/p/mR31F_gOIG. It seems the encoding/xml package's Unmarshal function, for a struct field like: <pre>Zs []int `xml:"x>y>z"`</pre> parses both `<x><y><z>0</z><z>1</z></y></x>` and `<x><y><z>0</z></y><y><z>1</z></y></x>`  as []int{0,1} - slightly surprising to me for the latter case , but useful nevertheless and more flexible. However, xml.Marshal will serialize `Zs: []int{0,1}` only as `<x><y><z>0</z><z>1</z></y></x>` - see for example http://play.golang.org/p/L2DkdvaFR2. Because SecurityGroupInfo type is also used by the ec2test server to generate the DescribeSecurityGroups response, the generated XML will in fact be *invalid* according to AWS XML schema (ipRanges>item>cidrIp,cidrIp rather than ipRanges>item>cidrIp,item>cidrIp). Go on the other hand handles both the valid and invalid XML the same way.

In summary, I realized *only* the ec2test server needs to change how it serializes security groups and the ec2 client code can continue to use []string for SourceIPs. So I'll revert the changes around SourceIPs and I'm sorry for being stupid :)